proxy/src/test/java/com/wavefront/agent/PreprocessorRulesTest.java

```diff
@@ -0,0 +1,274 @@
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testInvalidPointLineRule3() {
+    // try to create a whitelist rule with a null match pattern
+    PointLineWhitelistRegexFilter invalidRule = new PointLineWhitelistRegexFilter(null, null);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testInvalidPointLineRule4() {
+    // try to create a blacklist rule with a null match pattern
+    PointLineBlacklistRegexFilter invalidRule = new PointLineBlacklistRegexFilter(null, null);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testInvalidReportPointRule1() {
+    // try to create a blacklist rule with a null scope
+    ReportPointBlacklistRegexFilter invalidRule = new ReportPointBlacklistRegexFilter(null, "foo", null);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testInvalidReportPointRule2() {
+    // try to create a blacklist rule with a null pattern
+    ReportPointBlacklistRegexFilter invalidRule = new ReportPointBlacklistRegexFilter("foo", null, null);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testInvalidReportPointRule3() {
+    // try to create a whitelist rule with a null scope
+    ReportPointWhitelistRegexFilter invalidRule = new ReportPointWhitelistRegexFilter(null, "foo", null);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testInvalidReportPointRule4() {
+    // try to create a blacklist rule with a null pattern
+    ReportPointWhitelistRegexFilter invalidRule = new ReportPointWhitelistRegexFilter("foo", null, null);
+  }
+
+  @Test
+  public void testPointLineRules() {
+    String testPoint1 = "collectd.cpu.loadavg.1m 7 1459527231 source=hostname foo=bar boo=baz";
+    String testPoint2 = "collectd.#cpu#.&loadavg^.1m 7 1459527231 source=source$hostname foo=bar boo=baz";
+
+    PointLineReplaceRegexTransformer rule1 = new PointLineReplaceRegexTransformer("(boo)=baz", "$1=qux", null);
+    PointLineReplaceRegexTransformer rule2 = new PointLineReplaceRegexTransformer("[#&\\$\\^]", "", null);
+    PointLineBlacklistRegexFilter rule3 = new PointLineBlacklistRegexFilter(".*source=source.*", null);
+    PointLineWhitelistRegexFilter rule4 = new PointLineWhitelistRegexFilter(".*source=source.*", null);
+
+    String expectedPoint1 = "collectd.cpu.loadavg.1m 7 1459527231 source=hostname foo=bar boo=qux";
+    String expectedPoint2 = "collectd.cpu.loadavg.1m 7 1459527231 source=sourcehostname foo=bar boo=baz";
+
+    assertEquals(expectedPoint1, rule1.apply(testPoint1));
+    assertEquals(expectedPoint2, rule2.apply(testPoint2));
+    assertTrue(rule3.apply(testPoint1));
+    assertFalse(rule3.apply(testPoint2));
+    assertFalse(rule4.apply(testPoint1));
+    assertTrue(rule4.apply(testPoint2));
+  }
+
+  @Test
+  public void testReportPointRules() {
+    Map<String, String> annotations = new TreeMap<>();
```

Comment:
Fixed - went a different route (parse the point from a string instead of creating it on the fly)